pkg/controller/toolchainstatus/toolchainstatus_controller.go

```diff
@@ -176,6 +184,35 @@ func (r *ReconcileToolchainStatus) aggregateAndUpdateStatus(reqLogger logr.Logge
 
 	// if any components were not ready then set the overall status to not ready
 	if len(unreadyComponents) > 0 {
+		// If the current ToolchainStatus:
+		// a) Is currently not ready,
+		// b) has not been ready for longer than the configured threshold, and
+		// c) no notification has been already sent, then
+		// send a notification to the admin mailing list
+		c, found := condition.FindConditionByType(toolchainStatus.Status.Conditions, toolchainv1alpha1.ConditionReady)
+		if found && c.Status == corev1.ConditionFalse {
+			threshold := time.Now().Add(-minutesAfterUnready * time.Minute)
+			if c.LastUpdatedTime.Before(&metav1.Time{threshold}) {
```

Comment:
We wanna check `LastTransitionTime` here. Not `LastUpdatedTime`, do we?
`LastUpdatedTime` is updated every 5 seconds no matter if there are any real updates in the status. It's used as a "ping" timestamp, so we know if it's not updated then there is no running controller to monitor/update the status.
So, if we have our status controller running then you will never hit this 10 second treshshold. 

`LastTransitionTime` is updated only if there is a some real change in the condition (Ready True -> Ready False or Ready False -> Ready True). And this is exactly what we want to use to see if we stuck with Readh/False for more than 10 seconds, right?

Reply:
Good catch, fixed.